src/Promitor.Core.Scraping/Configuration/Serialization/Core/MetricsDeserializer.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using Promitor.Core.Scraping.Configuration.Model;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Promitor.Core.Scraping.Factories;
+using YamlDotNet.RepresentationModel;
+
+namespace Promitor.Core.Scraping.Configuration.Serialization
+{
+    internal class MetricsDeserializer : Deserializer<MetricDefinition>
+    {
+        internal MetricsDeserializer(ILogger logger) : base(logger)
+        {
+        }
+
+        internal override MetricDefinition Deserialize(YamlMappingNode node)
+        {
+            var rawResourceType = node.Children[new YamlScalarNode("resourceType")];
+
+            if (Enum.TryParse<ResourceType>(rawResourceType.ToString(), out var resourceType))
```

Comment:
How about throwing an exception when this guy returns false and get rid of the else?